### PR TITLE
Golang win

### PR DESCRIPTION
**Pull request by paul-lalonde**
A win written in Go.  This is a step to getting win working on Windows.  This will require a windows port of the pty package we're using.

### CODE REVIEW

**Review by rjkroege — APPROVED** [review]
Seems fine but maybe reformat the notes?

File: cmd/win/event.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+eventWhen a window’s event file is open, changes to the window occur as always but the actions are also reported as messages to the reader of the file. Also, user actions with buttons 2 and 3 (other than chorded Cut and Paste, which behave normally) have no immediate effect on the window; it is expected that the program reading the event file
```
[the remark this answers is not in the document]
Comment (paul-lalonde) [reply cmd/win/event.txt:1]: I copy-pasted this from acme(4) for quick reference; the man page is even worse in formatting.  I'll give it  bit of love.

File: cmd/win/win.go [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+func (w *winWin) israw() bool {
+	return (!w.cook && w.password) && !isecho(w.rcpty)
+}
+
+// Add text, either in the body, or at the not-yet-sent insertion point
+// The text might be in the message, or may need to come from
+// the data stream
+func (w *winWin) typetext(e *acme.Event) {
+	debugf("typetext %v @ w.p=%d\n", eToS(e), w.p)
+	buf := make([]byte, 256) // TODO(PAL): Don't want to allocate this every time!
```
[the remark this answers is not in the document]
Comment (paul-lalonde) [reply cmd/win/win.go:90]: Done.

**Review by rjkroege — APPROVED** [review]

**Comment by rjkroege** [comment]
This is awesome!

**Pull request merged**